### PR TITLE
Polkadot companion for Substrate#10463

### PR DESCRIPTION
**Pull request by acatangiu**
Companion for https://github.com/paritytech/substrate/pull/10463

Include the chain uid to the Grandpa protocol name.
Include the chain uid to the Beefy protocol name.
Related to https://github.com/paritytech/substrate/issues/7252 and https://github.com/paritytech/substrate/issues/7746 and https://github.com/paritytech/grandpa-bridge-gadget/issues/240.

### CODE REVIEW

**Review by ordian — APPROVED** [review]

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -755,10 +755,24 @@ where
 	// Note: GrandPa is pushed before the Polkadot-specific protocols. This doesn't change
 	// anything in terms of behaviour, but makes the logs more consistent with the other
 	// Substrate nodes.
-	config.network.extra_sets.push(grandpa::grandpa_peers_set_config());
-
+	let grandpa_protocol_name = grandpa::protocol_standard_name(
+		&client.block_hash(0).ok().flatten().unwrap_or_default(),
```
Comment (bkchr) [inline node/service/src/lib.rs:759]: Same as in Substrate

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -755,10 +755,24 @@ where
...
+		&config.chain_spec,
+	);
+	config
+		.network
+		.extra_sets
+		.push(grandpa::grandpa_peers_set_config(grandpa_protocol_name.clone()));
+
+	let beefy_protocol_name = beefy_gadget::protocol_standard_name(
+		&client.block_hash(0).ok().flatten().unwrap_or_default(),
```
Comment (bkchr) [inline node/service/src/lib.rs:768]: Dito

File: node/service/src/lib.rs (hunk 1 above)
Comment (acatangiu) [reply node/service/src/lib.rs]: done

**Comment by acatangiu** [comment]
@bkchr addressed comments, ptal

**Review by bkchr — APPROVED** [review]

**Comment by pepyakin** [comment]
The latest master have `continuous-integration/gitlab-zombienet-tests-parachains-smoke-test` failing. It apparently started with this PR. Can you, @acatangiu, look into that?

**Comment by acatangiu** [comment]
> The latest master have `continuous-integration/gitlab-zombienet-tests-parachains-smoke-test` failing. It apparently started with this PR. Can you, @acatangiu, look into that?

Sure, I'll look into it. How do I run that test? :smile:

**Comment by pepyakin** [comment]
TBH, I have no idea 🤷  Maybe @pepoviola could help with that?

**Comment by ordian** [comment]
I suspect #4668 fixes the issue (to be confirmed).

**Comment by pepoviola** [comment]
@ordian, yes looks like an issue with the image version. Also notice that the `malus` test are no using a fixed image of the collator. I don't know if you also want to update that test.
Thanks!

**Comment by ordian** [comment]
It does, but it's not clear why.
According to https://github.com/paritytech/substrate/pull/10463#issuecomment-1006166639, the change is backwards compatible for collators as well. Were there any breaking changes other than this PR in substrate maybe?

**Pull request merged**